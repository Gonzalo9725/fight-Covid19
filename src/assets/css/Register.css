.container-register{
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    height: -webkit-fill-available;
    background-image: url('../img/fondo.jpg');
    background-size: cover;

}

.container-register form{
    display: flex;
    flex-direction: column;
}

.register-form{
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    padding: 20px 50px;
    border-radius: 38px;
    background: #e9ebee;
    width: 25em;
    height: 25em;
}

.register-form h1{
    font-size: 2.1rem;
    margin-bottom: 1em;
}

.container-register form input{
    margin-bottom: 1em;
    width: 16em;
    height: 2.3em;
    border: none;
    background-color: #e9ebee;
    border-bottom: 1px solid #3c9699;
}

.container-register form button{
    margin-top: 2em;
}

@media (max-width: 500px) {

    .register-form h1{
        font-size: 1.8rem;
    }

    .container-register form input{
        width: 15em;
        height: 2.2em;
    }

    .register-form{
        width: 20em;
    }
}

@media (max-width: 350px) {

    .register-form h1{
        font-size: 1.6rem;
    }

    .container-register form input{
        width: 12em;
        height: 2em;
    }

    .register-form{
        width: 17em;
    }
}