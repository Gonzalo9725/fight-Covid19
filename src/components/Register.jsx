import React, { useState } from 'react'
import { auth } from '../firebase-config'
import { useHistory } from 'react-router-dom'
import '../assets/css/Register.css'

const Register = () => {

    const [ email, setEmail ] = useState('');
    const [ password, setPassword ] = useState('');
    const [ error, setError ] = useState(null);

    const history = useHistory();

    const signIn = async (e) => {
        e.preventDefault();

        if(!email.trim()){
            setError('Ingrese Email');
            return
        }

        if(!password.trim()){
            setError('Ingrese Password')
            return
        }

        if(password.length < 6){
            setError('La contraseña debe contener más de 6 caracteres')
            return
        }
        setError(null);
        createAccount()
    }

    const createAccount = async() => {
        try{
            await auth.createUserWithEmailAndPassword(email, password)
            alert('Se ha creado correctamente la cuenta');
            history.push('/')
        }catch (error) {
            if(error.code === 'auth/email-already-in-use'){
                setError('Email ya esta registrado')
            }
        }
    }

    return (
        <div className='container-register'>
            <div className='register-form'>
                <h1>Crear cuenta</h1>
                <form onSubmit={signIn}>
                    {
                        error && (
                            <div style={{color:'darkred'}}>
                                {error}
                            </div>
                        )
                    }
                    <input
                        id='mail'
                        onChange={(e) => setEmail(e.target.value)} 
                        type='email'
                        placeholder='Ingresa email...' 
                        value={email}/>
                    <input 
                        id='pass'
                        onChange={(e) => setPassword(e.target.value)}
                        type='password'
                        placeholder='Ingresa contraseña'
                        value={password}/>

                    <button className='register-button'>Registrarse</button>
                </form>
            </div>
        </div>
    )
}

export default Register