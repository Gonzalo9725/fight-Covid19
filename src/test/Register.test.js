import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Register from '../components/Register'

describe('Componente Register', () => {
    const form = mount( <Register/> );
    const testInput = {
        target: {
            value: 'gonzalovaras@gmail.com'
        }
    }

    const testInputPassword = {
        target: {
            value: '662211'
        }
    }

    test('Llamar OnChange al ingresar email', () => {
        form.find('#mail').simulate('change', testInput)
        expect(form.find('#mail').prop('value')).toEqual('gonzalovaras@gmail.com')
    })

    test('Llamar OnChange al ingresar password', () => {
        form.find('#pass').simulate('change', testInputPassword)
        expect(form.find('#pass').prop('value')).toEqual('662211')
    })

    test('El email debe ser un string', () => {
        form.find('#mail').simulate('change', testInput)
        expect(typeof(form.find('#mail').prop('value'))).toBe('string')
    })

    test('renders correctly', () => {
        const tree = renderer.create(<Register />).toJSON();
        expect(tree).toMatchSnapshot();
    });
})