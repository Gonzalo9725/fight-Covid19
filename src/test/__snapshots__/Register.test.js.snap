// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente Register renders correctly 1`] = `
<div
  className="container-register"
>
  <div
    className="register-form"
  >
    <h1>
      Crear cuenta
    </h1>
    <form
      onSubmit={[Function]}
    >
      <input
        id="mail"
        onChange={[Function]}
        placeholder="Ingresa email..."
        type="email"
        value=""
      />
      <input
        id="pass"
        onChange={[Function]}
        placeholder="Ingresa contraseÃ±a"
        type="password"
        value=""
      />
      <button
        className="register-button"
      >
        Registrarse
      </button>
    </form>
  </div>
</div>
`;
